crates/services/txpool/src/containers/dependency.rs

```diff
@@ -461,7 +442,14 @@ impl Dependency {
             // collision of other outputs is not possible.
         }
 
-        Ok((max_depth, db_coins, db_contracts, db_messages, collided))
+        Ok((
```

Comment:
The name is `check_for_collision`. I'd expect a function named that to return `Result<()>` or maybe `Result<Vec<TxId>>, but it's returning a bunch of `HashMap`s as well. Ideally we don't have single function responsible for multiple things, but at the very least we could name the function something more accurate.

I'd prefer to move the db extension logic to separate functions.

Reply:
I just did it the same way it had worked before. I don't see a point in refactoring the code that will be removed soon during the refactoring of the TxPool.